cloud_function/main.py

```diff
@@ -0,0 +1,127 @@
+import os
+import json
+import requests
+from google.cloud import storage
+from google.cloud import exceptions
+
+
+project = os.environ.get("GCP_PROJECT")
+service_account = os.environ.get('FUNCTION_IDENTITY')
+wfl_url = os.environ.get("WFL_URL")
+default_credentials = os.environ.get("DEFAULT_CREDENTIALS")
+
+
+def get_test_credentials():
+    import google.auth
+    import google.auth.transport.requests
+
+    scopes = ["https://www.googleapis.com/auth/cloud-platform",
+              "https://www.googleapis.com/auth/userinfo.email",
+              "https://www.googleapis.com/auth/userinfo.profile"]
+    credentials, gcp_project = google.auth.default(scopes=scopes)
+    if not credentials.valid:
+        credentials.refresh(google.auth.transport.requests.Request())
+    return credentials
+
+def get_auth_headers():
+    scopes_list = ["https://www.googleapis.com/auth/cloud-platform",
+                   "https://www.googleapis.com/auth/userinfo.email",
+                   "https://www.googleapis.com/auth/userinfo.profile"]
+    scopes = ",".join(scopes_list)
+    metadata_url = f'http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token?scopes={scopes}'
+    metadata_headers = {'Metadata-Flavor': 'Google'}
+    r = requests.get(metadata_url, headers=metadata_headers)
+    r.raise_for_status()
+    access_token = r.json()['access_token']
+    headers = {
+        'Authorization': 'Bearer {}'.format(access_token)
+    }
+    return headers
+
+def get_manifest_path(object_name):
+    path = object_name.strip("/")
+    chipwell_barcode, analysis_version, file_name = path.split("/", maxsplit=2)
+    return "/".join([chipwell_barcode, analysis_version, "ptc.json"])
+
+def get_or_create_workload(headers, cromwell_url):
+    payload = {
+        "creator": service_account,
+        "cromwell": cromwell_url,
+        "input": "aou-inputs-placeholder",
+        "output": "aou-ouputs-placeholder",
+        "pipeline": "AllOfUsArrays",
+        "project": "gotc-dev",
```

Comment:
@tbl3rd @rexwangcc would this field be in the "ptc.json" file or should we set it as an environment variable when deploying the function? 🤔

Reply:
This should be able to use the ptc.json as the request itself.

Reply:
I thought the contents of `ptc.json` were for the `/api/v1/append_to_aou` endpoint? This is to create/get the workload to update.

Reply:
You're right.

Reply:
Ok! So I replaced the hard-coded project here with the "environment" defined in ptc.json